receiver/k8sclusterreceiver/config.go

```diff
@@ -39,6 +39,11 @@ type Config struct {
 
 	// MetricsBuilderConfig allows customizing scraped metrics/attributes representation.
 	metadata.MetricsBuilderConfig `mapstructure:",squash"`
+
+	// Namespace to fetch resources from. If this is set, certain cluster-wide resources such as Nodes or Namespaces
+	// will not be able to be observed. Setting this option is recommended for sidecar deployment patterns where the collector runs
+	// as a sidecar for a pod, or in environments where due to security restrictions the collector can not be granted cluster-wide permissions.
+	Namespace string `mapstructure:"namespace"`
```

Comment:
I wonder if this effect should also be logged and/or be reflected to the affected resources' settings.

It would be weird to have the following:

```yaml
  k8s_cluster:
    auth_type: kubeConfig
    node_conditions_to_report: [Ready, MemoryPressure]
    namespace: staging
``` 

So would that make sense to ensure that affected metrics and settings are explicitly disabled in case the `namespace` is defined?

Maybe that's an overkill though so I would rely on code-owner's opinions here :)

Reply:
That's a good point, thanks @ChrsMark - I think it does make sense to check for this during the config validation, I'll add a check for that :)